libbeat/common/event.go

```diff
@@ -262,16 +262,22 @@ func DeDot(s string) string {
 // DeDotJSON replaces in keys all . with _
 // This helps when sending data to Elasticsearch to prevent object and key collisions.
 func DeDotJSON(json interface{}) interface{} {
-	switch json.(type) {
+	switch json := json.(type) {
 	case map[string]interface{}:
 		result := map[string]interface{}{}
-		for key, value := range json.(map[string]interface{}) {
+		for key, value := range json {
+			result[DeDot(key)] = DeDotJSON(value)
+		}
+		return result
+	case MapStr:
```

Comment:
Thanks for adding this option here. Could you add a test case for this "switch" option?

Reply:
@ruflin Tests added, and also made some changes to original test code...